modules/cnf-programming-the-n3000-operator.adoc

```diff
@@ -0,0 +1,351 @@
+N3000-daemonset-5k55l                       1/1         Running         1           15d
+N3000-discovery-blmjl                       1/1         Running         1           15d
+N3000-discovery-lblh7                       1/1         Running         1           15d
+----
+The following section provides information on the installed pods:
+* `fpga-driver-daemonset` provides and loads the required Open Programmable Accelerator Engine (OPAE) drivers
+* `fpgainfo-exporter` provides N3000 telemetry data for Prometheus
+* `N3000-controller-manager` applies N3000Node CRs to the cluster and manages all the operand containers
+* `N3000-daemonset` is the main worker application
+* `N3000-discovery` discovers N3000 Accelerator devices installed and labels worker nodes if devices are present
+
+. Get all the nodes containing the Intel® FPGA PAC N3000 card:
++
+[source,terminal]
+----
+$ oc get n3000node
+----
++
+.Example output
+[source,terminal]
+----
+NAME                       FLASH
+node1                      NotRequested
+----
+
+. Retrieve information about the card on each node:
++
+[source,terminal]
+----
+$ oc get n3000node node1 -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+status:
+  conditions:
+  - lastTransitionTime: "2020-12-15T17:09:26Z"
+    message: Inventory up to date
+    observedGeneration: 1
+    reason: NotRequested
+    status: "False"
+    type: Flashed
+  fortville:
+  - N3000PCI: 0000:1b:00.0
+    NICs:
+    - MAC: 64:4c:36:11:1b:a8
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1a:00.0
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+    - MAC: 64:4c:36:11:1b:a9
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1a:00.1
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+    - MAC: 64:4c:36:11:1b:ac
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1c:00.0
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+    - MAC: 64:4c:36:11:1b:ad
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1c:00.1
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+  fpga:
+  - PCIAddr: 0000:1b:00.0 <1>
+    bitstreamId: "0x23000410010310" <2>
+    bitstreamVersion: 0.2.3
+    deviceId: "0x0b30"
+----
+<1> Specify the PCI address of the card to be programmed.
+<2> This is the currently flashed bitstream.
+
+. Save the current `bitstreamId`, `PCIAddr`, the name, and the `deviceId` without "0x" padding.
++
+[source,terminal]
+----
+# oc get n3000node -o json
+----
+
+. Update the user bitstream of the Intel® FPGA PAC N3000 card user:
+
+.. Create a custom resource (CR) containing information about what node and card should be programmed together with the bitstream you want to flash the card with. Save the YAML in the `n3000-cluster.yaml` file.
```

Comment:
Using the pattern from the first instruction, I think you can snip this down:

```
Define the N3000 cluster resource to program by creating a file named `n3000-cluster.yaml` as shown in the following example:
```

All the details are spelled out in callouts so I think you can leave the instructional text a little more spare.